lib/tasks.js

```diff
@@ -0,0 +1,397 @@
+'use strict';
+
+const utils = require('./utils');
+const Timer = require('./timer');
+const Task = require('./task');
+const noop = cb => cb();
+
+/**
+ * Static factory method for creating a custom `Composer` class that
+ * extends the given `Emitter`.
+ *
+ * ```js
+ * const Emitter = require('events');
+ * const Composer = require('composer').create(Emitter);
+ * const composer = new Composer();
+ * ```
+ * @name .create
+ * @param {function} `Emitter` Event emitter.
+ * @return {Class} Returns a custom `Composer` class.
+ * @api public
+ */
+
+module.exports = Emitter => {
```

Comment:
Should this be validated or at least default to the internal emitter: `module.exports = (Emitter = require('./events')) => {`

Reply:
No, `task` should not have a customizable emitter. We need to make sure that task doesn't get, say, `Templates` or `Assemble` as the Emitter, which would cause major issues.